compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaUnionTypeSymbol.java

```diff
@@ -84,9 +102,10 @@ public String signature() {
             return "...";
         }
         List<TypeSymbol> memberTypes = this.memberTypeDescriptors();
-        if (memberTypes.size() == 2 &&
-                memberTypes.get(1).typeKind() == TypeDescKind.NIL) {
-            return memberTypes.get(0).signature() + "?";
+        if (memberTypes.size() == 2) {
+            TypeSymbol member1 = memberTypes.get(0);
+            return member1.typeKind() == TypeDescKind.NIL ?
+                    memberTypes.get(1).signature() + "?" : member1.signature() + "?";
```

Comment:
Is this correct, what if T1|T2 where T2 is not nil,

Reply:
Fixed this with e53c44b